### PR TITLE
added saltwater module

### PR DESCRIPTION
**Pull request by trinemykk**
connected to #1245 in opm-common and #367 in opm_material

### CODE REVIEW

File: opm/models/blackoil/blackoilsaltwatermodules.hh [hunk 1]
```diff
@@ -0,0 +1,427 @@
+// -*- mode: C++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*-
```
Comment (akva2) [inline opm/models/blackoil/blackoilsaltwatermodules.hh]: need to add this file to the CMakeLists_files.txt installation list.

File: opm/models/blackoil/blackoilsaltwatermodules.hh [hunk 2]
```diff
@@ -0,0 +1,427 @@
...
+  copyright holders.
+*/
+/*!
+ * \file
+ *
+ * \brief Contains the classes required to extend the black-oil model by SALTwater.
+ */
+#ifndef EWOMS_BLACK_OIL_SALTWATER_MODULE_HH
+#define EWOMS_BLACK_OIL_SALTWATER_MODULE_HH
+
```
Comment (totto82) [inline opm/models/blackoil/blackoilsaltwatermodules.hh]: BRINE_MODULE

File: opm/models/blackoil/blackoilsaltwatermodules.hh [hunk 3]
```diff
@@ -0,0 +1,427 @@
...
+        if (!enableSaltWater)
+            // saltwaters have been disabled at compile time
+            return false;
+
+        return pvIdx == saltConcentrationIdx;
+    }
+
+    /*!
+     * \brief Assign the solvent specific primary variables to a PrimaryVariables object
+     */
```
Comment (totto82) [inline opm/models/blackoil/blackoilsaltwatermodules.hh]: copy past typo

File: opm/models/blackoil/blackoilsaltwatermodules.hh [hunk 4]
```diff
@@ -0,0 +1,427 @@
...
+
+        priVars[saltConcentrationIdx] = fluidState.saltconcentration();
+    }
+
+    static std::string primaryVarName(unsigned pvIdx)
+    {
+        assert(primaryVarApplies(pvIdx));
+
+        return "saltwater_waterconcentration";
+    }
```
Comment (totto82) [inline opm/models/blackoil/blackoilsaltwatermodules.hh]: -> saltconcentration

File: opm/models/blackoil/blackoilsaltwatermodules.hh [hunk 5]
```diff
@@ -0,0 +1,427 @@
...
+        if (!enableSaltWater)
+            return;
+
+        const auto& extQuants = elemCtx.extensiveQuantities(scvfIdx, timeIdx);
+
+        const unsigned upIdx = extQuants.upstreamIndex(FluidSystem::waterPhaseIdx);
+        const unsigned inIdx = extQuants.interiorIndex();
+        const auto& up = elemCtx.intensiveQuantities(upIdx, timeIdx);
+        //const unsigned contiWaterEqIdx = Indices::conti0EqIdx + Indices::canonicalToActiveComponentIndex(FluidSystem::waterCompIdx);
+
```
Comment (totto82) [inline opm/models/blackoil/blackoilsaltwatermodules.hh]: This can be removed

**Pull request closed without merging**